tests/Testcontainers.Dapr.Tests/DaprContainerTest.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Dapr.Client;
+using System;
+
+namespace Testcontainers.Dapr;
+
+public sealed class DaprContainerTest : IAsyncLifetime
+{
+    private DaprContainer _daprContainer;
+
+    public Task InitializeAsync()
+    {
+        _daprContainer = new DaprBuilder()
+            .WithAppId("testicorns")
+            .WithLogLevel("debug")
+            .Build();
+
+        return _daprContainer.StartAsync();
+    }
+
+    public Task DisposeAsync()
+    {
+        return _daprContainer.DisposeAsync().AsTask();
+    }
+
+    [Fact]
+    [Trait(nameof(DockerCli.DockerPlatform), nameof(DockerCli.DockerPlatform.Linux))]
+    public async Task CheckDaprSideCarIsHealthy()
+    {            
+        Environment.SetEnvironmentVariable("DAPR_HTTP_PORT", _daprContainer.DaprHttpPort.ToString());
+        Environment.SetEnvironmentVariable("DAPR_GRPC_PORT", _daprContainer.DaprGrpcPort.ToString());
```

Comment:
Can we please provide a connection string from `DaprContainer`? The host is not always `localhost`. Depending on the Docker environment, it could be something else.

```csharp
_ = new UriBuilder(Uri.UriSchemeHttp, Hostname, GetMappedPublicPort(DaprBuilder.DaprHttpPort)).ToString();
_ = new UriBuilder(Uri.UriSchemeHttp, Hostname, GetMappedPublicPort(DaprBuilder.DaprGrpcPort)).ToString();
```

Reply:
Hmmmm. 

I'll think on this, as I think the Dapr Client library is hardcoded to always use `localhost`

https://github.com/dapr/dotnet-sdk/blob/b97af45c5f2689ecfd85246b5954b6bc2f528e07/src/Shared/DaprDefaults.cs#L79

Reply:
The `DaprClientBuilder` offers `UseHttpEndpoint(string)` and `UseGrpcEndpoint(string)`, both receives an URI endpoint. I guess that's the right way to configure the Daper client. I would be surprised if only `localhost` is supported (also only IPv4).

Reply:
Dapr is mostly used in a kubernetes environments, where each k8s service has its own dapr sidecar within the pod, so overriding the host (`27.0.0.1`) is quite rare/unnecessary tbh.

However, sure, I can include the connection string for posterity.

Reply:
I've addressed this in my latest changes:

https://github.com/testcontainers/testcontainers-dotnet/blob/bf4bcbb32758d90113ea76087c80088d0e8815f0/src/Testcontainers.Dapr/DaprContainer.cs#L17-L26

It is important to understand that connections from the **test host** requires to resolve `Hostname` and the random assigned host port (`GetMappedPublicPort(ushort))`. While a container-to-container communication does not. They can simply use the network alias and port to connect to (**note**: they connect straight to the container port).